internal/flex/flex.go

```diff
@@ -78,6 +78,13 @@ func FlattenFloat64Pointer(f *float64) types.Float64 {
 	return FlattenFloat64(*f)
 }
 
+func FlattenFloat32Pointer(f *float32) types.Float64 {
```

Comment:
FYI, newer plugin framework version supports `types.Float32`, which should help us avoid such casts. 

https://github.com/hashicorp/terraform-plugin-framework/releases/tag/v1.10.0

Reply:
Would it be ok to update Plugin Framework in this PR itself ?

Reply:
can you please do in a different pr?

Reply:
Updating plugin framework might require proper testing . Could it be done after this one ,  we can remove the typecasting for Int32 as well .